diff --git a/src/main/java/net/pitan76/itemalchemy/tile/EMCCollectorTile.java b/src/main/java/net/pitan76/itemalchemy/tile/EMCCollectorTile.java
index 116934f..8a8bc0c 100644
--- a/src/main/java/net/pitan76/itemalchemy/tile/EMCCollectorTile.java
+++ b/src/main/java/net/pitan76/itemalchemy/tile/EMCCollectorTile.java
@@ -89,8 +89,10 @@ public void tick(TileTickEvent<EMCCollectorTile> e) {
 
         if (coolDown == 0) {
             if (isFull()) return;
-            float skyAngle = world.getSkyAngle(0);
-            if ((!WorldUtil.isRaining(world) && !WorldUtil.isThundering(world) && (WorldUtil.hasSkyLight(world) && skyAngle <= 0.25 || skyAngle >= 0.75) && WorldUtil.isSkyVisible(world, pos.up()))
+            
+            long time = world.getTimeOfDay() % 24000L;
+            boolean isDay = time < 12000L;
+            if ((!WorldUtil.isRaining(world) && !WorldUtil.isThundering(world) && (WorldUtil.hasSkyLight(world) && isDay) && WorldUtil.isSkyVisible(world, pos.up()))
                     || WorldUtil.getBlockState(world, pos.up()).getLuminance() > 10 || WorldUtil.getBlockState(world, pos.down()).getLuminance() > 10
                     || WorldUtil.getBlockState(world, pos.north()).getLuminance() > 10 || WorldUtil.getBlockState(world, pos.south()).getLuminance() > 10
                     || WorldUtil.getBlockState(world, pos.east()).getLuminance() > 10 || WorldUtil.getBlockState(world, pos.west()).getLuminance() > 10) {
